crates/chia-sdk-driver/src/primitives/streamed_cat.rs

```diff
@@ -0,0 +1,448 @@
+use crate::{CatLayer, DriverError, Layer, Puzzle, Spend, SpendContext};
+use chia_consensus::gen::make_aggsig_final_message::u64_to_bytes;
+use chia_protocol::{Bytes, Bytes32, Coin};
+use chia_puzzle_types::{
+    cat::{CatArgs, CatSolution},
+    CoinProof, LineageProof,
+};
+use chia_sdk_types::{run_puzzle, Condition, Conditions};
+use chia_sha2::Sha256;
+use clvm_traits::FromClvm;
+use clvm_utils::tree_hash;
+use clvmr::{op_utils::u64_from_bytes, Allocator, NodePtr};
+
+use crate::{StreamLayer, StreamPuzzleSolution};
+
+#[derive(Debug, Clone)]
+#[must_use]
+pub struct StreamedCat {
+    pub coin: Coin,
+    pub asset_id: Bytes32,
+    pub proof: LineageProof,
+    pub inner_puzzle_hash: Bytes32,
```

Comment:
Is there a way to turn this into `StreamedCatInfo` and calculate the inner puzzle hash on the fly? Similar to `DidInfo`